File: CircularPrimes/CP.py
#!/usr/bin/env python3
"""
Project Euler # 35

"""

from sympy import isprime
from math import sqrt

def sieve(n): # Sieve of Eratosthenes
    x = [False, False] + [True] * (n-2)

    for i in range(2, int(sqrt(n))+1):
        if x[i] is True:
            for j in range(i*2, n, i):
                x[j] = False

    return enumerate(x, 0)

def circular(n): # Finds out if prime is circular
    x = str(n)
    for i in range(1, len(x)):
        a = x[i::] + x[:i:]
        if not isprime(int(a)):
            return False
    return True

print(sum(1 for i, j in sieve(10**6) if j and circular(i)))

File: CircularPrimes/CP.java
/*
Project Euler # 35

*/

public class CP {
    public static void main( String[] args ) {
      boolean[] primes = sieve(1000000);
      int count = 0;


      for ( int i=0; i<primes.length; i++ ) {
          if ( primes[i] == true ) {
              String n = Integer.toString(i);
              boolean flag = true;
              for ( int j=1; j<n.length(); j++ )
                  if ( !primes[rotate(n, j)] ) {
                      flag = false;
                      break;
                  }
              if ( flag )
                  count++;
          }
      }

      System.out.println(count);
    }

    public static boolean[] sieve( int n ) {
        boolean[] primes = new boolean[n+1];
        primes[0] = false;
        primes[1] = false;

        for( int i=2; i<primes.length; i++ )
            primes[i] = true;

        for( int i=2; i<=Math.sqrt(n)+1; i++ )
            if( primes[i] == true )
                for ( int j=i*i; j<=n; j+=i)
                    primes[j] = false;

        return primes;
    }

    public static int rotate(String s, int offset ) {
        int i = offset % s.length();
        s = s.substring(i) + s.substring(0, i);
        return Integer.parseInt(s);
    }



}
